### PR TITLE
CI: Upgrade to the latest spock framework

### PR DESCRIPTION
**Pull request by gavin-stackrox**
## Description

Upgrades to the latest spock+groovy. The main change is that `@Category` is now `@Tag` which got rid of the whole Groups.XYZ thing.

## Checklist
- [x] Investigated and inspected CI test results

## Testing Performed

More testing:
- [x] default [BAT](https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/stackrox_stackrox/4135/pull-ci-stackrox-stackrox-master-gke-qa-e2e-tests/1603854196564561920/artifacts/gke-qa-e2e-tests/stackrox-e2e/build-log.txt)
- [x] all-tests
- [x] [upgrade](https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/stackrox_stackrox/4135/pull-ci-stackrox-stackrox-master-gke-upgrade-tests/1603854196589727744/artifacts/gke-upgrade-tests/stackrox-e2e/build-log.txt)
- [x] compat
- [x] openshift
- [x] `-race`

### CODE REVIEW

**Comment by janisz** [comment]
There are some issues with building images on OSCI. I added `turbo-build` label to build it on GHA

File: qa-tests-backend/src/test/groovy/IntegrationsSplunkViolationsTest.groovy [hunk 1]
```diff
@@ -101,7 +99,8 @@ class IntegrationsSplunkViolationsTest extends BaseSpecification {
                 ["name": SPLUNK_INPUT_NAME, "interval": "1", "from_checkpoint": "2000-01-01T00:00:00.000Z"])
     }
 
-    @Category([Integration, BAT])
+    @Tag("Integration")
+    @Tag("BAT")
```
Comment (RTann) [inline qa-tests-backend/src/test/groovy/IntegrationsSplunkViolationsTest.groovy:103]: probably doesn't matter, but typically, I have seen BAT first

File: qa-tests-backend/src/test/groovy/IntegrationsTest.groovy [hunk 2]
```diff
@@ -37,10 +34,9 @@ import util.MailServer
 import util.SplunkUtil
 
 import org.junit.Assume
-import org.junit.AssumptionViolatedException
```
Comment (RTann) [inline qa-tests-backend/src/test/groovy/IntegrationsTest.groovy:40]: was this unused?

**Review by RTann — APPROVED** [review]

File: qa-tests-backend/build.gradle [hunk 3]
```diff
@@ -160,6 +148,81 @@ test {
     }
 }
 
+task testBegin(type: Test) {
+    useJUnitPlatform {
+        includeTags "Begin"
+    }
+}
+
+task testRest(type: Test) {
```
[the remark this answers is not in the document]
Comment (gavin-stackrox) [reply qa-tests-backend/build.gradle:157]: testRest==the former 'all' default test task but now minus 'begin'.

File: qa-tests-backend/src/test/groovy/ComplianceTest.groovy [hunk 4]
```diff
@@ -57,10 +55,9 @@ import services.RoleService
 import util.Timer
 
 import org.junit.Assume
-import org.junit.experimental.categories.Category
 import spock.lang.IgnoreIf
-import spock.lang.Requires
```
[the remark this answers is not in the document]
Comment (gavin-stackrox) [reply qa-tests-backend/src/test/groovy/ComplianceTest.groovy:62]: @Requires still works it just cannot make API calls. I guess in the 1.0 version it ran after the BaseSpec had done that globalSetup().

File: qa-tests-backend/src/test/groovy/IntegrationsTest.groovy (hunk 2 above)
Comment (gavin-stackrox) [reply qa-tests-backend/src/test/groovy/IntegrationsTest.groovy:40]: It is in a commented out test. I should have mentioned in the PR description, I ran the IntelliJ 'autofix imports'. Which after the global removal of groups.* seemed like the easiest way to 'clean up' the imports but it also created a lot of shuffling which adds noise to an already noisy PR :(
We could do with a one time application of an opinionated style to these Groovies IMO but I will hold off :)

File: qa-tests-backend/src/test/groovy/IntegrationsSplunkViolationsTest.groovy (hunk 1 above)
Comment (gavin-stackrox) [reply qa-tests-backend/src/test/groovy/IntegrationsSplunkViolationsTest.groovy:103]: > There are some issues with building images on OSCI. I added `turbo-build` label to build it on GHA

Thanks! That got the images built in time for the e2e tests 🕺

**Pull request merged**